src_c/base.c

```diff
@@ -1924,6 +1924,14 @@ static PyObject *
 pg_GetDefaultWindowSurface(void)
 {
     /*return a borrowed reference*/
+    if (pg_default_window) {
+        /* With SDL2, resizing invalidates the existing surface.  Calling
+         * SDL_GetWindowSurface will recreate the surface when it is
+         * invalid (or return the current surface if it is still valid)
+         */
+        SDL_Surface *new_surface = SDL_GetWindowSurface(pg_default_window);
```

Comment:
SDL_GetWindowSurface can return NULL on failure

Reply:
Ok.  My initial take on that would be to raise an `pgExc_SDLError` here if `SDL_GetWindowSurface` returns NULL.  However, I am not sure how to do that in base.c because the "standard" method for this does not work (`return RAISE(pgExc_SDLError, SDL_GetError());`).

Notably, neither `pgExc_SDLError` nor `PYGAMEAPI_GET_SLOT` is defined here (presumably to avoid breaking `base.c` that end up doing the true definition of `pgExc_SDLError` later)

Reply:
Hrmm.

Here's the docs for pg_GetDefaultWindowSurface.
```
.. c:function:: PyObject* pg_GetDefaultWindowSurface(void)

   Return a borrowed reference to the Pygame default window display surface,
   or *NULL* if no default window is open.

   Availability: SDL 2.
```

It doesn't talk about errors. I guess all the callers will have to check for the `surf` value of the return being NULL being an error - and then raise an exception? 

```c
PyObject *surface = pg_GetDefaultWindowSurface();
if(surface && !surface->surf) {
    RAISE(pgExc_SDLError, SDL_GetError());
}
```

It's kind of an error prone API though, but would work. What do you think?

Reply:
I don't think SDL_GetWindowSurface will fail that often. We can move this `check SDL_GetWindowSurface return code` issue to a separate issue report for another time... and merge this one in.

Reply:
I wrote this up here: https://github.com/pygame/pygame/issues/1671